### PR TITLE
Fix darkmode toggle

### PR DESCRIPTION
**Pull request by R055A**
## Description

remove error message displayed on browser

### Type of change

- [x] Bug fix (added a non-breaking change which fixes an issue)

## How Has This Been Tested?

- [x] Ran tests with `composer test`

## Checklist:

- [x] I have checked to make sure no other pull requests are open for this issue
- [x] The code is properly formatted and is consistent with the existing code style
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/demo/index.php [hunk 1]
```diff
@@ -26,7 +26,9 @@ function gtag() { dataLayer.push(arguments); }
     <script async defer src="https://buttons.github.io/buttons.js"></script>
 </head>
 
-<body <?php echo $_COOKIE["darkmode"] == "on" ? 'data-theme="dark"' : ""; ?>>
+<body <?php if (!empty($_COOKIE)) {
+        echo $_COOKIE["darkmode"] == "on" ? 'data-theme="dark"' : "";
+} ?>>
```
Comment (DenverCoder1) [inline src/demo/index.php:31]: isset should be used to check, and can keep it on 1 line

```suggestion
<?php echo (isset($_COOKIE["darkmode"]) && $_COOKIE["darkmode"] == "on") ? 'data-theme="dark"' : ""; ?>
```

**Review by DenverCoder1 — COMMENTED** [review]
Good catch! Thanks for fixing this.

**Comment by DenverCoder1** [comment]
May want to fix it here as well

image

File: src/demo/index.php (hunk 1 above)
Comment (R055A) [reply src/demo/index.php:31]: Do you want me to do this?

Comment (R055A) [reply src/demo/index.php]: I was going to include isset(). Removing the is not empty if-condition removes the fix so I have made a new commit with both.

**Comment by DenverCoder1** [comment]
If you have `isset`, you shouldn't need `!empty` also

File: src/demo/index.php [hunk 2]
```diff
@@ -26,7 +26,7 @@ function gtag() { dataLayer.push(arguments); }
     <script async defer src="https://buttons.github.io/buttons.js"></script>
 </head>
 
-<body <?php echo $_COOKIE["darkmode"] == "on" ? 'data-theme="dark"' : ""; ?>>
+<?php echo (isset($_COOKIE["darkmode"]) && !empty($_COOKIE) && $_COOKIE["darkmode"] == "on") ? 'data-theme="dark"' : ""; ?>
```
Comment (DenverCoder1) [inline src/demo/index.php:29]: body tag was removed

```suggestion
<body <?php echo (isset($_COOKIE["darkmode"]) && $_COOKIE["darkmode"] == "on") ? 'data-theme="dark"' : ""; ?>>
```

**Comment by R055A** [comment]
> 
> 
> If you have `isset`, you shouldn't need `!empty` also

isset() doesn't resolve the bug.

File: src/demo/index.php [hunk 3]
```diff
@@ -115,7 +115,7 @@ function gtag() { dataLayer.push(arguments); }
     </div>
 
     <a href="javascript:toggleTheme()" class="darkmode" title="toggle dark mode">
-        <i class="<?php echo $_COOKIE["darkmode"] == "on" ? 'gg-sun' : "gg-moon"; ?>"></i>
+        <i class="<?php echo (isset($_COOKIE["darkmode"]) && !empty($_COOKIE) && $_COOKIE["darkmode"] == "on") ? 'gg-sun' : "gg-moon"; ?>"></i>
```
Comment (DenverCoder1) [inline src/demo/index.php:118]: isset should be enough, don't need both

```suggestion
        <i class="<?php echo (isset($_COOKIE["darkmode"]) && $_COOKIE["darkmode"] == "on") ? 'gg-sun' : "gg-moon"; ?>"></i>
```

**Comment by DenverCoder1** [comment]
> > If you have `isset`, you shouldn't need `!empty` also
> 
> isset() doesn't resolve the bug.

I pretty much always do that since I think it's the usual way to do it. I will double check :+1:

**Comment by DenverCoder1** [comment]
Either of these should be fine

```php
isset($_COOKIE['notice'])
```

```php
!empty($_COOKIE['notice'])
```

https://stackoverflow.com/a/20206033/11608064

I think isset() is easier to understand in this context.

File: src/demo/index.php (hunk 2 above)
Comment (R055A) [reply src/demo/index.php]: Sorry, I copied the suggested change from the prior review change request comment. It seems to be correct now

File: src/demo/index.php (hunk 3 above)
Comment (R055A) [reply src/demo/index.php]: Done

**Comment by DenverCoder1** [comment]
Does it work for you? I've tested it and it looks fine.

**Review by DenverCoder1 — APPROVED** [review]

File: src/demo/index.php (hunk 2 above)
Comment (DenverCoder1) [reply src/demo/index.php]: Oh, I guess that was my mistake earlier, then

**Comment by DenverCoder1** [comment]
Awesome! Thanks for your contribution! :tada:

**Pull request merged**